projects/igniteui-angular/src/lib/data-operations/data-util.ts

```diff
@@ -30,24 +30,31 @@ export enum DataType {
  * @hidden
  */
 export class DataUtil {
-    public static sort<T>(data: T[], expressions: ISortingExpression[], sorting: IgxSorting = new IgxSorting()): T[] {
+    public static sort<T>(data: T[], expressions: ISortingExpression[], sorting?: IGridSortingStrategy): T[] {
+        if (!sorting) {
+            sorting = new IgxSorting();
+        }
         return sorting.sort(data, expressions);
     }
 
     public static treeGridSort(hierarchicalData: ITreeGridRecord[],
         expressions: ISortingExpression[],
-        parent?: ITreeGridRecord): ITreeGridRecord[] {
+        parent?: ITreeGridRecord,
+        sorting?: IGridSortingStrategy): ITreeGridRecord[] {
```

Comment:
Swap the parent and sorting parameters, so that the parent is the last parameter.

Reply:
Fixed.